### PR TITLE
Fixed hover not working in VR

### PR DESCRIPTION
**Pull request by sinnstudio-nas**
Added a check in hover event for VR

### CODE REVIEW

File: Source/UINavigation/Private/UINavWidget.cpp [hunk 1]
```diff
@@ -3105,7 +3109,7 @@ void UUINavWidget::HoverEvent(int Index)
 		return;
 	}
 
-	if (UINavPC->GetCurrentInputType() != EInputType::Mouse || Index == ButtonIndex)
+	if (Index == ButtonIndex || !UHeadMountedDisplayFunctionLibrary::IsHeadMountedDisplayEnabled() && (UINavPC->GetCurrentInputType() != EInputType::Mouse && (!bUseLeftThumbstickAsMouse || !UINavPC->IsMovingLeftStick())))
```
Comment (goncasmage1) [inline Source/UINavigation/Private/UINavWidget.cpp:3112]: You need to separate the IsHeadMountedDisplay boolean into its own local variable, so you don't have to call it if you're not using a VR platform

**Review by goncasmage1 — CHANGES_REQUESTED** [review]
Be sure to merge this into the 4.24 branch 😁

File: Source/UINavigation/Private/UINavWidget.cpp (hunk 1 above)
Comment (sinnstudio-nas) [reply Source/UINavigation/Private/UINavWidget.cpp]: Fixed.

File: Source/UINavigation/Private/UINavWidget.cpp [hunk 2]
```diff
@@ -3105,7 +3109,12 @@ void UUINavWidget::HoverEvent(int Index)
 		return;
 	}
 
-	if (UINavPC->GetCurrentInputType() != EInputType::Mouse || Index == ButtonIndex)
+	bool bIsVR = false;
+	#if IS_VR_PLATFORM 
+	bIsVR = UHeadMountedDisplayFunctionLibrary::IsHeadMountedDisplayEnabled();
+	#endif
+
+	if (Index == ButtonIndex || !bIsVR && (UINavPC->GetCurrentInputType() != EInputType::Mouse && (!bUseLeftThumbstickAsMouse || !UINavPC->IsMovingLeftStick())))
```
Comment (goncasmage1) [inline Source/UINavigation/Private/UINavWidget.cpp:3117]: Put the !bIsVR inside the parenthesis

**Review by goncasmage1 — CHANGES_REQUESTED** [review]

**Review by goncasmage1 — APPROVED** [review]

**Pull request merged**